src/ray/util/event.cc

```diff
@@ -189,16 +189,25 @@ void EventManager::Publish(const rpc::Event &event, const json &custom_fields) {
 }
 
 void EventManager::PublishExportEvent(const rpc::ExportEvent &export_event) {
-  for (const auto &element : reporter_map_) {
-    (element.second)->ReportExportEvent(export_event);
+  auto element = export_log_reporter_map_.find(export_event.source_type());
```

Comment:
can you assert this always exist? or is there a case it doesn't?

Reply:
yes, the reporter should always exist in the map

Reply:
btw you can do
```
RAY_CHECK(export_log_reporter_map_ != export_log_reporter_map_.end());
(element->second)->ReportExportEvent(export_event); 
```

can you handle this in your next PR?